pkg/controller/logstash/volume/volumes.go

```diff
@@ -0,0 +1,114 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package volume
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+
+	commonv1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/common/v1"
+	logstashv1alpha1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/logstash/v1alpha1"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/volume"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/utils/set"
+)
+
+// AppendDefaultPVCs appends defaults PVCs to a set of existing ones.
+//
+// The default PVCs are not appended if:
+// - a Volume with the same .Name is found in podSpec.Volumes, and that volume is not a PVC volume
+func AppendDefaultPVCs(existing []corev1.PersistentVolumeClaim, ls logstashv1alpha1.Logstash) []corev1.PersistentVolumeClaim {
+	// create a set of volume names that are not PVC-volumes
+	existingVolumes := set.Make()
+
+	for _, existingPVC := range existing {
+		existingVolumes.Add(existingPVC.Name)
+	}
+
+	for _, volume := range ls.Spec.PodTemplate.Spec.Volumes {
+		existingVolumes.Add(volume.Name)
+	}
+
+	for _, defaultPVC := range DefaultVolumeClaimTemplates {
+		if existingVolumes.Has(defaultPVC.Name) {
+			continue
+		}
+		existing = append(existing, defaultPVC)
+	}
+	return existing
+}
+
+func BuildVolumesAndMounts(ls logstashv1alpha1.Logstash) ([]corev1.Volume, []corev1.VolumeMount) {
+	persistentVolumes := make([]corev1.Volume, 0)
+
+	// Add Default logs volume to list of persistent volumes
+	persistentVolumes = append(persistentVolumes, DefaultLogsVolume)
```

Comment:
`DefaultLogsVolume` is EmptyDir which is shared in pod's lifetime. The name `persistentVolumes` is a bit confusing. Suggest to rename to `volumes`.